ui/lib/core/addon/components/replication-page.js

```diff
@@ -20,128 +23,123 @@ import { task } from 'ember-concurrency';
     }
-    this.set('reindexingDetails', resp);
-  }),
-  isSummaryDashboard: computed('model.{performance.mode,dr.mode}', function () {
-    const router = this.router;
-    const currentRoute = router.get('currentRouteName');
+  }
+  get isSummaryDashboard() {
+    const currentRoute = this.router.currentRouteName;
 
     // we only show the summary dashboard in the replication index route
     if (currentRoute === 'vault.cluster.replication.index') {
-      const drMode = this.model.dr.mode;
-      const performanceMode = this.model.performance.mode;
+      const drMode = this.args.model.dr.mode;
+      const performanceMode = this.args.model.performance.mode;
       return drMode === 'primary' && performanceMode === 'primary';
     }
     return '';
-  }),
-  formattedReplicationMode: computed('model.replicationMode', 'isSummaryDashboard', function () {
+  }
+  get formattedReplicationMode() {
     // dr or performance 🤯
     const { isSummaryDashboard } = this;
     if (isSummaryDashboard) {
       return 'Disaster Recovery & Performance';
     }
-    const mode = this.model.replicationMode;
+    const mode = this.args.model.replicationMode;
     return MODE[mode];
-  }),
-  clusterMode: computed('model.replicationAttrs', 'isSummaryDashboard', function () {
+  }
+  get clusterMode() {
     // primary or secondary
-    const { model } = this;
-    const { isSummaryDashboard } = this;
-    if (isSummaryDashboard) {
+    if (this.isSummaryDashboard) {
       // replicationAttrs does not exist when summaryDashboard
       return 'primary';
     }
-    return model.replicationAttrs.mode;
-  }),
-  isLoadingData: computed('clusterMode', 'model.replicationAttrs', function () {
-    const { clusterMode } = this;
-    const { model } = this;
-    const { isSummaryDashboard } = this;
+    return this.args.model.replicationAttrs.mode;
+  }
+  get isLoadingData() {
+    const { isSummaryDashboard, clusterMode } = this;
     if (isSummaryDashboard) {
       return false;
     }
-    const clusterId = model.replicationAttrs.clusterId;
-    const replicationDisabled = model.replicationAttrs.replicationDisabled;
+    const clusterId = this.args.model.replicationAttrs.clusterId;
+    const replicationDisabled = this.args.model.replicationAttrs.replicationDisabled;
     if (clusterMode === 'bootstrapping' || (!clusterId && !replicationDisabled)) {
       // if clusterMode is bootstrapping
       // if no clusterId, the data hasn't loaded yet, wait for another status endpoint to be called
       return true;
     }
     return false;
-  }),
-  isSecondary: computed('clusterMode', function () {
+  }
+  get isSecondary() {
     const { clusterMode } = this;
     return clusterMode === 'secondary';
-  }),
-  replicationDetailsSummary: computed('isSummaryDashboard', function () {
-    const { model } = this;
+  }
+  get replicationDetailsSummary() {
     const { isSummaryDashboard } = this;
     if (!isSummaryDashboard) {
-      return;
+      return {};
     }
```

Comment:
nit: can we remove this block since an empty object is the default return below?

Reply:
an undefined response was making something else fail unless `isSummaryDashboard` was true, so this was intentional. Ideally I'd like this page to not do so much 😃 but baby steps